src/rdkb/impl.c

```diff
@@ -0,0 +1,543 @@
+		WebcfgDebug("Get_Webconfig_URL. retPsmGet %d tempUrl %s\n", retPsmGet, tempUrl);
+		if (retPsmGet == RBUS_ERROR_SUCCESS)
+                {
+			if(tempUrl !=NULL)
+			{
+				strcpy(pString , tempUrl);
+			}
+			WebcfgInfo("Inside cpeabs lib file\n");
+			WebcfgDebug("Get_Webconfig_URL. pString %s\n", pString);
+		}
+		else
+                {
+                        WebcfgError("psm_get failed ret %d for parameter %s\n", retPsmGet, WEBCFG_URL_PARAM);
+                }
+	}
+	WebcfgDebug("Get_Webconfig_URL strong fn from lib\n");
+	return retPsmGet;
+}
+
+int Set_Webconfig_URL( char *pString)
+{
+    int retPsmSet = 0;
+    if(isRbusEnabled())
+    {
+    	retPsmSet = rbus_StoreValueIntoDB( WEBCFG_URL_PARAM, pString );
+	WebcfgDebug("Set_Webconfig_URL. retPsmSet %d pString %s\n", retPsmSet, pString);
+
+	if (retPsmSet != RBUS_ERROR_SUCCESS)
+        {
+		WebcfgError("psm_set failed ret %d for parameter %s and value %s\n", retPsmSet, WEBCFG_URL_PARAM, pString);
+                return 0;
+        }
+        else
+        {
+		WebcfgInfo("Inside cpeabs lib file\n");
+		WebcfgDebug("psm_set success ret %d for parameter %s and value %s\n", retPsmSet, WEBCFG_URL_PARAM, pString);
+        }
+    }
+    return retPsmSet;
+}
+
+int Get_Supplementary_URL( char *name, char *pString)
+{
+	char *tempUrl = NULL;
+	int retPsmGet = 0;
+	if(isRbusEnabled())
+	{
+		char *tempParam = (char *) malloc (sizeof(char)*MAX_BUFF_SIZE);
+		if(tempParam !=NULL)
+		{
+			snprintf(tempParam, MAX_BUFF_SIZE, "%s%s", WEBCFG_PARAM_SUPPLEMENTARY_SERVICE, name);
+			WebcfgDebug("tempParam is %s\n", tempParam);
+			retPsmGet = rbus_GetValueFromDB( tempParam, &tempUrl);
+			if (retPsmGet == RBUS_ERROR_SUCCESS)
+			{
+				WebcfgInfo("Inside cpeabs lib file\n");
+				WebcfgDebug("Get_Supplementary_URL. retPsmGet %d tempUrl %s\n", retPsmGet, tempUrl);
+				if(tempUrl !=NULL)
+				{
+					strcpy(pString , tempUrl);
+				}
+				WebcfgDebug("Get_Supplementary_URL. pString %s\n", pString);
+				WEB_FREE(tempParam);
+			}
+			else
+			{
+				WebcfgError("psm_get failed ret %d for parameter %s\n", retPsmGet, tempParam);
+				WEB_FREE(tempParam);
+			}
+		}
+	}
+	return retPsmGet;
+}
+
+int Set_Supplementary_URL( char *name, char *pString)
+{
+    int retPsmSet = 0;
+    if(isRbusEnabled())
+    {
+	char *tempParam = (char *) malloc (sizeof(char)*MAX_BUFF_SIZE);
+	if(tempParam !=NULL)
+	{
+		if ((name != NULL) && (strncmp(name, "Telemetry",strlen(name)+1)) == 0)
+		{
+			snprintf(tempParam, MAX_BUFF_SIZE, "%s%s", WEBCFG_PARAM_SUPPLEMENTARY_SERVICE, name);
+			WebcfgDebug("tempParam is %s\n", tempParam);
+			retPsmSet = rbus_StoreValueIntoDB( tempParam, pString );
+			if (retPsmSet != RBUS_ERROR_SUCCESS)
+			{
+				WebcfgError("psm_set failed ret %d for parameter %s%s and value %s\n", retPsmSet, WEBCFG_PARAM_SUPPLEMENTARY_SERVICE, name, pString);
+				WEB_FREE(tempParam);
+				return 0;
+			}
+			else
+			{
+				WebcfgInfo("Inside cpeabs lib file\n");
```

Comment:
If testing is completed, we can disable this info logs

Reply:
will remove once this new review comment changes were added and tested